.github/workflows/main.yml

```diff
@@ -23,31 +23,32 @@ jobs:
           path: .
           fetch-depth: 0  # avoid shallow clone with no tags
 
-      - name: Install Mamba
-        run: |
-          wget "https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh" -O mambaforge.sh
-          bash mambaforge.sh -b -p $HOME/mambaforge
-          echo "$HOME/mambaforge/bin" >> $GITHUB_PATH
-          source $HOME/mambaforge/etc/profile.d/conda.sh
-          conda init bash
+      - name: initialize miniconda
+        # this uses a marketplace action that sets up miniconda in a way that makes
+        # it easier to use.  I tried setting it up without this and it was a pain
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: test
+          # environment.yml file is needed by this action.  Because I don't want
+          # maintain this but rather maintain the requirements files it just has
+          # basic things in it like conda and pip
+          environment-file: ./environment.yml
+          python-version: 3
+          auto-activate-base: false
 
-      - name: Create conda environment
+      - name: install diffpy.pdfgui requirements
+        shell: bash -l {0}
         run: |
-          source $HOME/mambaforge/etc/profile.d/conda.sh
-          conda create -n test_env python=3.12 xonsh regolith -c conda-forge -y
-          conda activate test_env
-          pip install case_insensitive_dict
-
-      - name: Install dependencies
+          conda config --set always_yes yes --set changeps1 no
+          conda config --add channels conda-forge
+          conda activate test
+          conda install --file requirements/run.txt
+          conda install --file requirements/test.txt
+          pip install .
+      - name: Validate diffpy.pdfgui
+        shell: bash -l {0}
         run: |
-          source $HOME/mambaforge/etc/profile.d/conda.sh
-          conda activate test_env
-          mamba install --file requirements/run.txt
-          mamba install --file requirements/test.txt
-          pip install -e .
-
-      - name: Use Xvfb Action
```

Comment:
we will need this xvfb action for the wx.   

This PR is getting the tests to pass and will take some work, so put all the workflow work on a separate PR.   I had it working before with the mamba stuff and we may need to do it that way, but let's try and get as far as we can with the cookiecutter version, then see if we can the xvfb action to work.  WHen I was doing it before, it couldn't activate the conda environment for some reason.  Let's work on this on a separate PR.